Library/Homebrew/formula_installer.rb

```diff
@@ -242,7 +243,9 @@ def install
     build_bottle_preinstall if build_bottle?
 
     unless @poured_bottle
-      compute_and_install_dependencies if @pour_failed && !ignore_deps?
+      if (!pour_bottle || @pour_failed) && !ignore_deps?
```

Comment:
Could you pull `!pour_bottle || @pour_failed` into a separate variable (because binary logic hurts my brain)